### ISSUE TITLE
Option to store my posts in github

### ISSUE THREAD

**Issue by schollii** [thepracticaldev/dev.to#7840]
**Is your feature request related to a problem? Please describe.**

Content I write is on dev.to. It would be nice for the markdown to be in my own storage, like github. Really the blog/post hosting service should just be the presentation layer (styling, layout, navigation, etc) and community features. 

A way to store my dev.to posts on github, if I set it up that way:

- Namely I could write the post on dev.to but it would actually create a PR on my github repo. Once the PR is merged, the post becomes publicly visible on dev.to. If I edit the post in dev.to after PR merged, same process. The github repo setting would be repo URL, and dev.to would put the markdown and images etc in there in whatever fashion it desires. 

- Comments on post: If the post has changed since a comment was written, the comment would show the git hash of the post; clicking on it would show the post as it appeared for that hash. Also, on the post in dev.to would probably be best put in the commentor's github, if they have that configured, but that's a nice-to-have.

- The rendered posts would be cached so that even if the github repo is not reachable (due to a temporary service downtime by github, or other issue), the posts are visible. 

- If for some reason I edit the markdown in git repo directly, I'm leaning towards dev.to not doing anything about it, ie no monitoring/webhooks/etc. Eg if I write a post on dev.to and it ends up with commit HASH A1, and then I edit the markdown on my desktop and I push a new commit A2, dev.to still shows A1... why would it not, I think I should have to use the dev.to web app (or REST API) to decide what goes on dev.to. I suppose it would be convenient if I could go to the post, click on Update to Commit, and it would show the list of commits since the page's current commit, I could select one. 

Such capability could also pave the road to enabling users to suggest modifications to a page (this would create a PR etc). 

Using your download feature, unzip it in my git clone, pushing it. This is not nearly as convenient as I have to remember to do that. 

This is **not** the same as me writing posts on my desktop then pushing to github and dev.to gets notified to render. I think this would be significantly more involved than what I propose, although it would be interesting too. The solution above is simpler to manage because dev.to still controls what goes in my repo. 

I'd be happy to work on this.

**Comment by citizen428** [comment]
Thanks for the issue! We'll take your request into consideration and follow up if we decide to tackle this issue.

To our amazing contributors: issues labeled `type: bug` are always up for grabs, but for feature requests, please wait until we add a `ready for dev` before starting to work on it.

To claim an issue to work on, please leave a comment. If you've claimed the issue and need help, please ping @thepracticaldev/oss and we will follow up within 3 business days.

For full info on how to contribute, please check out our [contributors guide](https://github.com/thepracticaldev/dev.to/blob/master/CONTRIBUTING.md).

**Comment by joshpuetz** [comment]
I'm not sure if this would address your concerns, but publishing from a static GitHub site to DEV via RSS feeds is something we support. I do this with my own blog: it's hosted on GitHub Pages (https://github.com/joshpuetz/joshpuetz.github.io), and I publish on DEV via RSS. It's certainly not as full featured as what you've proposed, but it is a work around!

**Comment by schollii** [comment]
@joshpuetz thanks for sharing that idea, I have never setup RSS, can you point to one of your posts on dev.to that was pulled that way (and the link to the post in github), I'd like to see what that looks like. And what's involved in setting this up this way. As you, I would prefer what I proposed but your solution may be a useful workaround while waiting for this to happen ;)

**Comment by joshpuetz** [comment]
Sure thing @schollii, a little background: I'm serving my site on GitHub pages, and I use Jekyll to generate posts. 

Here's a post I made on my blog:
https://joshpuetz.com/2020/03/18/Thoughts-on-announced-2020-ipad-pro-and-magic-keyboard.html

Here's the source in GitHub:
https://github.com/joshpuetz/joshpuetz.github.io/blob/master/_posts/2020-03-18-Thoughts-on-announced-2020-ipad-pro-and-magic-keyboard.md

And here's the post on Dev:
https://dev.to/joshpuetz/thoughts-on-announced-2020-ipad-pro-and-magic-keyboard-3845

Now that I'm mentioning it, RSS import support is quite limited: only a few front matter tags are imported (notably things like a header image aren't imported). Here's the source code: 
https://github.com/thepracticaldev/dev.to/blob/master/app/services/rss_reader.rb#L69-L80
